backend/handler/AuthHandler.ts

```diff
@@ -8,44 +7,80 @@ import { getLogger } from './LoggerHandler';
 
 const logger = getLogger('AuthHandler');
 
-let authorizationCache = {};
+let authorizationCache = {} as ConfigResponse['authorization'];
 
-export function getToken(basicAuth) {
-    return jsonRequest('GET', '/tokens', {
+export interface TokenResponse {
+    username: string;
+    value: string;
+    role: string;
+}
+
+export interface ConfigResponse {
+    metadata: any;
+    items: any[];
+    authorization: {
+        roles: {
+            id: number;
+            name: string;
+            type: string;
+            description: string;
+        }[];
+        permissions: Record<string, string[]>[];
+    };
+}
+
+export interface VersionResponse {
+    edition: typeof EDITION.PREMIUM | typeof EDITION.COMMUNITY;
+    version: string;
+    build: any;
+    date: any;
+    commit: any;
+    distribution: string;
+    // eslint-disable-next-line camelcase
+    distro_release: string;
+}
+
+export function getToken(basicAuth: string) {
+    return jsonRequest<TokenResponse>('GET', '/tokens', {
         Authorization: basicAuth
     });
 }
 
-export function getTenants(token) {
+export function getTenants(token: string) {
     return jsonRequest('GET', '/tenants?_get_all_results=true&_include=name', {
         'Authentication-Token': token
     });
 }
 
-export function getUser(token) {
+export function getUser(token: string) {
     return jsonRequest('GET', '/user?_get_data=true', {
         'Authentication-Token': token
     });
 }
 
-export function isProductLicensed(version) {
+export function isProductLicensed(version: VersionResponse) {
     return !_.isEqual(version.edition, EDITION.COMMUNITY);
 }
 
-export function getLicense(token) {
+export function getLicense(token: string) {
     return jsonRequest('GET', '/license', {
         'Authentication-Token': token
     });
 }
 
-export function getTokenViaSamlResponse(samlResponse) {
-    return jsonRequest('POST', '/tokens', null, {
-        'saml-response': samlResponse
-    });
+export function getTokenViaSamlResponse(samlResponse: string) {
+    return jsonRequest<TokenResponse>(
+        'POST',
+        '/tokens',
+        {},
+        {
+            'saml-response': samlResponse
+        }
+    );
 }
 
-export function getAndCacheConfig(token) {
-    return jsonRequest('GET', '/config', {
+export function getAndCacheConfig(token: string | undefined) {
```

Comment:
Optional:
We can shorten that typing with a `?`:  syntax

```suggestion
export function getAndCacheConfig(token?: string) {
```

Reply:
Sure, changed.